### ISSUE TITLE
关于CISPO中mask问题

### ISSUE THREAD

**Issue by Yshihui** [MiniMax-AI/MiniMax-M1#17]
- 论文中讲到，如果A>0 且r>1+x时，直接mask掉不进行更新，这个跟论文中要解决”however“等特殊token在后几步增大，导致r超过上限被GRPO阶段的问题，是违背的，这个怎么理解这部分和GRPO的区别？

![Image](https://github.com/user-attachments/assets/3f9a9e62-c772-4d9c-b09f-035f00a72206)
另外，sg是只提供系数值而不保留r中对应的参数梯度更新能力吗？

**Comment by MiniMax-AI-Dev** [comment]
PPO 和 GRPO 中的 clip(, min()) 的操作等价于下面的 M_{i, t}，在标准的 CISPO 中把 \epsilon_low 和 \epsilon_high 都设置的很大，让这个 mask 不生效，只通过前面的 \epsilon^IS_low 来对 importance sampling weight 进行截断，保持对关键 token 的学习。
sg(r) 等价于 r = r.detach()

**Comment by Yshihui** [comment]
\epsilon_low 和 \epsilon_high 以及IS的\epsilon_low和 \epsilon_high 论文中没有体现，这两组值你们实验中最好的配置是多少？谢谢
